### ISSUE TITLE
TypeError on config run (ESM Support)

### ISSUE THREAD

**Issue by AverageHelper** [fi3ework/vite-plugin-checker#81]
I have my Vite config set up like so:

```ts
import { defineConfig } from "vite";
import vue from "@vitejs/plugin-vue";
import checker from "vite-plugin-checker";

export default defineConfig({
	plugins: [
		vue(),
		checker({
			typescript: true,
			vueTsc: true,
		}),
	],
});
```

I'm using this in a Vue 3 project, and my aim is to set up proper type checks separate from VSCode's [since Vite does not intend to support this, even at build time](https://github.com/vitejs/vite/issues/2539#issuecomment-809273298).

After adding the `vite-plugin-checker` import and calling `checker`, running `vite` or `vue-tsc` results in an error:

```sh
failed to load config from /[...]/vite.config.js
error when starting dev server:
TypeError: checker is not a function
```

As far as I can tell, I've imported `vite-plugin-checker` correctly according to the README documentation. Either I'm missing some obvious thing here, or the export structure is not the same at runtime as its declared type.

**Comment by philer** [comment]
I've got the same issue, would be grateful for a solution.

**Comment by vmarchaud** [comment]
I just had the issue, it was because i had configured `"type": "module"`  in your package.json that forces vite to use esm however `vite-plugin-checker` is still published as CJS

**Comment by buckeyez** [comment]
Any update on this? Is it on the agenda?

**Comment by fi3ework** [comment]
This issue will be fixed in days.

**Comment by fi3ework** [comment]
fixed in 0.4.5
